woocommerce/woocommerce-ios
# Shipping Labels: state selection in address validation screen

Closes #4464 

## Description
In this PR, I implemented the screen and the logic for display the list of states, tapping on the State field in the Shipping Label address validation. Plus, I disabled the interaction on the country field, since it will be editable in M4.

## Changes
- Handled a custom mapping case in the model `StateOfACountry` where, sometimes, the returned `code` it’s not a string but an integer.
- Fixed the wooApiVersion in `DataRemote` under the endpoint for loading the countries.
- Updated `WordPressKit` to version 4.37.0-beta.2 because of the conflicts with the `Country` model under WordPressKit.
- Fetch countries from the remote endpoint in Shipping Label Form.
- Implemented the `ListSelectorViewController` for display and select the states of a country.

## Testing
1. Make sure you have configured the WooCommerce WCShip plugin, and you have one or more payment methods set up in WP Admin under WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Launch the app in debug mode (because of the feature flag).
3. Open an order detail that is eligible for the shipping label.
4. Tap the button "Create Shipping Label".
5. Open an address, and make sure you are able to modify the `State` from the List Selector but not the `Country`.


## Screenshots
| Customer detail without note             |  Customer detail with note |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 12 Pro - 2021-06-30 at 17 45 14](https://user-images.githubusercontent.com/495617/123991407-fca19080-d9ca-11eb-9fd6-04c861b8e57e.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-06-30 at 17 45 18](https://user-images.githubusercontent.com/495617/123991438-00cdae00-d9cb-11eb-89ba-b21c984be3e6.png)



Update release notes:
